src/pray.c

```diff
@@ -2169,7 +2169,7 @@ doturn(void)
         if (!mtmp->mpeaceful
             && (is_undead(mtmp->data) || is_vampshifter(mtmp)
                 || (is_demon(mtmp->data) && (u.ulevel > (MAXULEV / 2))))) {
-            wakeup(mtmp, FALSE);
+            wakeup(mtmp, FALSE, FALSE);
```

Comment:
Not that this should be called for mimics anyway, but I think if there were like, a zombie mimic, and you turned it, it would be revealed, no? It has to be revealed to flee.